ocaml/xapi/xapi_host.ml

```diff
@@ -2629,14 +2639,34 @@ let get_sched_gran ~__context ~self =
 let emergency_disable_tls_verification ~__context =
   (* NB: the tls-verification state on this host will no longer agree with state.db *)
   Stunnel_client.set_verify_by_default false ;
-  Unixext.unlink_safe Xapi_globs.verify_certificates_path
+  Unixext.unlink_safe Xapi_globs.verify_certificates_path ;
+  try
+    (* we update the database on a best-effort basis because we
+       might not have a connection *)
+    let self = Helpers.get_localhost ~__context in
+    Db.Host.set_tls_verification_enabled ~__context ~self ~value:false
```

Comment:
So what actually happens here in practice when you have lost connection to the coordinator and you try to disable verification? I ask because I recall trying to make a similar sort of change in the past, and getting an indefinite hang rather than an exception

Reply:
(I'm thinking about the situation where the user has a genuine networking problem, but attempts to disable tls verification to see if it helps)

Reply:
I don't know. If networking breaks down and this is not related because of TLS problems, I think you have all kinds of problems, probably your VMs can't talk to storage anymore. Not sure we can guarantee anything at that point

Reply:
I now see failed unit tests - need to investigate

Reply:
The failure is coming from this change: